pkg/tekton/metapipeline/clientfactory.go

```diff
@@ -216,18 +228,21 @@ func versionStreamURLAndRef(jxClient versioned.Interface, ns string) (string, st
 		ref = config.DefaultVersionsRef
 	}
 
-	return url, ref, nil
+	return &teamSettings, url, ref, nil
 }
 
 func (c *clientFactory) cloneVersionStreamIfNeeded() error {
-	url, ref, err := versionStreamURLAndRef(c.jxClient, c.ns)
+	teamSettings, url, ref, err := versionStreamURLAndRef(c.jxClient, c.ns)
 	if err != nil {
 		return err
 	}
 
 	if c.versionStreamURL != url || c.versionStreamRef != ref {
 		oldVersionStreamDir := c.versionDir
-		c.versionDir, err = cloneVersionStream(url, ref)
+		c.versionDir, _, err = versionstreamrepo.CloneJXVersionsRepo(url, ref, teamSettings, c.git, true, false, c.stdin, c.stdout, c.stderr)
+		c.teamSettings = teamSettings
```

Comment:
You keep updating the instance `teamSetting`, but I cannot see where it is used.

Reply:
Ah, good point. I did need it at one point but then refactored.

Reply:
let's get rid of it then :-)